db/migrate/20250611083925_fix_stale_billing_entity_sequential_id_to_be_uniq.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+class FixStaleBillingEntitySequentialIdToBeUniq < ActiveRecord::Migration[8.0]
+  class Invoice < ApplicationRecord
+  end
+
+  class BillingEntity < ApplicationRecord
+  end
+
+  def up
+    # BillingEntities::ChangeInvoiceNumberingService -- if we switch from per_customer
+    # to per_billing_entity, we'll recalculate the billing_entity_sequential_id
+    BillingEntity.where(document_numbering: "per_customer").find_each do |billing_entity|
+      # find invoices with duplicated billing_entity_sequential_id
+      duplicates = Invoice.where(billing_entity_id: billing_entity.id)
+        .non_self_billed.with_generated_number
+        .where.not(billing_entity_sequential_id: nil)
+        .group(:billing_entity_sequential_id)
+        .having("COUNT(*) > 1")
+        .pluck(:billing_entity_sequential_id)
+      next if duplicates.empty?
+
+      # update the billing_entity_sequential_id to NULL for the duplicated invoices
+      # rubocop:disable Rails/SkipsModelValidations
+      Invoice.where(billing_entity_id: billing_entity.id)
+        .non_self_billed.with_generated_number
+        .where.not(billing_entity_sequential_id: nil)
```

Comment:
not needed as next line already defines the accepted values for billing_entity_sequential_id. self billed and invoices without generated number does not have billing entity sequential id by definition...

```suggestion
        .non_self_billed.with_generated_number
        .where.not(billing_entity_sequential_id: nil)
```